### PR TITLE
Enable Checksum csc param for mscorlib

### PR DESCRIPTION
**Pull request by joperezr**
cc: @gkhanna79

related: dotnet/corefx#8580

### CODE REVIEW

File: src/mscorlib/GenerateCompilerResponseFile.targets [hunk 1]
```diff
@@ -16,7 +16,8 @@
       <Clean Include="$(IntermediateOutputPath)\moduleName.rsp" />
     </ItemGroup>
     <PropertyGroup>
-      <CompilerResponseFile>$(IntermediateOutputPath)\moduleName.rsp</CompilerResponseFile>
+      <CompilerResponseFile Condition="'$(CompilerResponseFile)'!=''">$(IntermediateOutputPath)\moduleName.rsp;$(CompilerResponseFile)</CompilerResponseFile>
```
Comment (gkhanna79) [inline src/mscorlib/GenerateCompilerResponseFile.targets]: Can this be simplified to be something like this:

<CompilerResponseFile>$(IntermediateOutputPath)\moduleName.rsp;$(CompilerResponseFile)</CompilerResponseFile>

Also, does this change needs to account for System.Private.CoreLib that is also being built?

Comment (gkhanna79) [inline src/mscorlib/GenerateCompilerResponseFile.targets]: CC @AlexGhiondea

**Comment by joperezr** [comment]
@dotnet-bot test this please

File: src/mscorlib/GenerateCompilerResponseFile.targets (hunk 1 above)
Comment (joperezr) [inline src/mscorlib/GenerateCompilerResponseFile.targets]: I'll test to make sure that we can make that a one-liner as you suggest.

I tested with building the root build.proj project, and verified that all 4 projects(which includes System.Private.CoreLib) that actually invoke csc.exe are passing the checksum flag now.

Comment (AlexGhiondea) [inline src/mscorlib/GenerateCompilerResponseFile.targets]: @joperezr can you change the name of the response file from `modulename.rsp` to `modulename.$(AssemblyName).rsp`? 

That way there won't be a race condition between the build of mscorlib and the build of System.Private.CoreLib.

The race condition problem is going to go away as soon as we merge the rename change but changing the name of the response file to be unique per assembly is still worth it.

Comment (joperezr) [inline src/mscorlib/GenerateCompilerResponseFile.targets]: Done.

**Comment by joperezr** [comment]
@dotnet-bot test CentOS7.1 x64 Debug Build and Test please

**Comment by joperezr** [comment]
Merging now since build is currently broken and my change will fix it, plus I got a green CI before rebasing. Any additional feedback I'll address in my next PR.

**Pull request merged**